src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2042,13 +2052,48 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode,
 										 newTargetEntryListForSortClauses);
 	}
 
+	if (workerExtendedOpNode->windowClause)
```

Comment:
In general,  `WorkerExtendedOpNode ()` and `MasterExtendedOpNode ()` functions are becoming really, really long. 

Do you think would it make sense to refactor these functions into smaller ones? Say, can we move most this logic directly into another function (e.g., `List * GenerateNewTargetEntriesForWindowClause(windowClause, ...)` or smth like that)?

Reply:
we should do that as a part of window function pull. It too will have changes scattered to these two functions.